command/ibft/ibft_init.go

```diff
@@ -69,46 +93,141 @@ var (
 	libp2pDir    = "libp2p"
 )
 
+// setupLocalSM is a helper method for boilerplate local secrets manager setup
+func setupLocalSM(dataDir string) (secrets.SecretsManager, error) {
```

Comment:
ibft_init.go should have no concept of the implementation behind SecretsManager.

It beats the point to have a standardized interface, only to have specific Vault or Local secretmanager logic here.

Reply:
You're completely right 👍 

Fixed in 9ce1965 & 2b152a4